### DISCUSSION TITLE
Focus and crop on image don't work

### DISCUSSION THREAD

**Discussion by Romaixn** [sulu/sulu#6038] [Q&A]
| Q | A
| --- | ---
| Sulu Version | 2.2.7
| PHP Version | 7.4 / 8.0.3
| DB Version | PostgreSQL
| Browser Version | Firefox 89

# Actual Behavior

Freeze when click on "Set Focus" and "Crop image" link on media image page. The preview of image don't display and current user is disconnected.
![image](https://user-images.githubusercontent.com/18333296/117943133-1c420400-b30c-11eb-88e6-4300b61c8935.png)

# Expected Behavior

I expect focus and crop work well

**Comment by niklasnatter** [comment]
Hey,
I cannot reproduce this problem in my installation and on `sulu.rocks`. 
Do you see any errors in the development tools of your browser?

**Comment by Romaixn** [comment]
I juste have a 401 "Unauthorized" when trying to fetch image

**Comment by niklasnatter** [comment]
> I juste have a 401 "Unauthorized" when trying to fetch image

Its hard to help you with this amount of information. Can you please include screenshots of the failed requests and the errors in you javascript console?

Do the medias work in other parts of the system? Have you tested the problem in different environments? Does it happen in all of them? Do you see any errors in the logs (`var/logs/*`) of your application?

**Comment by Romaixn** [comment]
I have this error on var/logs/admin/prod.log : 
`request.CRITICAL: Uncaught PHP Exception Symfony\Component\ErrorHandler\Error\FatalError: "Compile Error: Declaration of Proxies\__CG__\Sulu\Bundle\MediaBundle\Entity\CollectionType::addCollection(Doctrine\Common\Collections\Collection $collections) must be compatible with Sulu\Bundle\MediaBundle\Entity\CollectionType::addCollection(Sulu\Bundle\MediaBundle\Entity\CollectionInterface $collections)" at /home/hero6927/public_html/var/cache/admin/prod/doctrine/orm/Proxies/__CG__SuluBundleMediaBundleEntityCollectionType.php line 252 {"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Compile Error: Declaration of Proxies\\__CG__\\Sulu\\Bundle\\MediaBundle\\Entity\\CollectionType::addCollection(Doctrine\\Common\\Collections\\Collection $collections) must be compatible with Sulu\\Bundle\\MediaBundle\\Entity\\CollectionType::addCollection(Sulu\\Bundle\\MediaBundle\\Entity\\CollectionInterface $collections) at /home/hero6927/public_html/var/cache/admin/prod/doctrine/orm/Proxies/__CG__SuluBundleMediaBundleEntityCollectionType.php:252)"} []`

and that in javascript console : 
![image](https://user-images.githubusercontent.com/18333296/117948269-44803180-b311-11eb-9072-9a63c05b3f95.png)

**Comment by niklasnatter** [comment]
Have you tried to clear the cache of your application by removing the `var/cache` folder?

Can you please answer the following questions asked above:

- Do the medias work in other parts of the system? 
- Have you tested the problem in different environments? Does it happen in all of them?

**Comment by Romaixn** [comment]
Yes, the problem persist.

The medias work well in other parts, I can add media and display them.

The problem does not appear in my local machine (Linux with PHP 8.0.6), Is a PHP module missing?

**Comment by alexander-schranz** [comment]
> The problem does not appear in my local machine

Sounds for me like a caching issues if cleaning the `var/cache` directory didn't work its mostly a hint that the class is in the `opcache` still cached. On some server you can just restart webserver or php-fpm. Else there is a tool called cachetool which can do this https://github.com/gordalina/cachetool as the webserver opcache can not be cleared over cli directly.
